From c49e620fadda9d572b7649a99b1c1423abf4f288 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 29 Oct 2024 08:26:34 +0100
Subject: [PATCH 1348/1695] clk: stm32mp25: remove critical clocks when not
 tdcid

Remove critical clocks are they are now managed on TDCID side.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c25a99e150f1b639c03e1605ed29d979a52023ad clk: stm32mp25: remove critical clocks when not tdcid]
Change-Id: Icdb738b2de11b0b4dbefd5549d4b7fd2998da5e0
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/417263
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index fa94e0c58..acae62aaa 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -4435,7 +4435,11 @@ static void clk_stm32_set_flexgen_as_critical(void)
 
 static bool clk_stm32_clock_is_critical(struct clk *clk)
 {
+
 	struct clk *clk_criticals[] = {
+#ifdef CFG_STM32_CM33TDCID
+		&ck_icn_p_gicv2m
+#else
 		&ck_hsi,
 		&ck_hse,
 		&ck_msi,
@@ -4475,6 +4479,7 @@ static bool clk_stm32_clock_is_critical(struct clk *clk)
 		&ck_icn_p_ipcc2,
 		&ck_icn_p_gicv2m,
 		&ck_icn_p_rtc
+#endif
 	};
 	size_t i = 0;
 
-- 
2.47.2

